eis_toolkit/spatial_analyses/cba.py

```diff
@@ -0,0 +1,511 @@
+
+    Compute spatial joint and generate the table of presence of the different
+    attributes.
+
+    Args:
+        geodata: the geodata object corresponding to the vector file.
+        disc: existing disk 'paving'.
+        target_attribut: target attribute to extract from the vector file.
+        subset_of_target_attribut_values: subset of the relevant values
+            of the target attribute
+
+    Returns: pair of variables
+        -> boolean indicating if the target attribute must be dummified
+        -> grid resulting from the intersection between the mesh and the
+            vector file
+    """
+    dummification, target, identified_values = check_and_prepare_param(
+        geodata, target_attribut, subset_of_target_attribut_values
+    )
+
+    # Spatial join of the intersection between the grid and the current map
+    join_grid = gpd.sjoin(disc, geodata, how="inner", predicate="intersects")
+
+    if dummification:
+        tmp = pd.get_dummies(join_grid[join_grid[target].isin(identified_values)][[target]], prefix="", prefix_sep="")
+        indicators = tmp.groupby(tmp.index).max()
+    else:
+        tmp = join_grid[target]
+        indicators = tmp.groupby(tmp.index).max()
+    return indicators, join_grid
+
+
+def get_grid(  # type: ignore[no-any-unimported]
+    working_map: gpd.GeoDataFrame, cell_width: int, cell_height: int
+) -> gpd.GeoDataFrame:
+    """Intermediate utility.
+
+    Produces a regular grid based on geodata surface.
+    Paving begin from the upper left corner (North-West) and go to the South
+    and East (up to down, left to right paving).
+
+    Args:
+        working_map: geodata used as a 'bounding box'.
+        cell_width: width of cells.
+        cell_height: height of cells.
+
+    Returns:
+        GeoDataFrame containing all the cells paving the geodata area.
+    """
+    xmin, ymin, xmax, ymax = working_map.total_bounds
+    rows = int(np.ceil((ymax - ymin) / cell_height))
+    cols = int(np.ceil((xmax - xmin) / cell_width))
+    x_left_origin = xmin
+    x_right_origin = xmin + cell_width
+    y_top_origin = ymax
+    y_bottom_origin = ymax - cell_height
+    print("Columns: %s" % cols)
+    print("Rows: %s" % rows)
+    print("Cells: %s" % (cols * rows))
+    data = []
+    for i in range(cols):
+        y_top = y_top_origin
+        y_bottom = y_bottom_origin
+        for j in range(rows):
+            polygon = Polygon(
+                [
+                    (x_left_origin, y_top),
+                    (x_right_origin, y_top),
+                    (x_right_origin, y_bottom),
+                    (x_left_origin, y_bottom),
+                ]
+            )
+            data += [[i + j * cols, i, j, polygon]]
+            y_top = y_top - cell_height
+            y_bottom = y_bottom - cell_height
+        x_left_origin = x_left_origin + cell_width
+        x_right_origin = x_right_origin + cell_width
+    result = pd.DataFrame(data, columns=["cell_id", "x", "y", "geom"])
```

Comment:
Unless there is a specific reason, I would not allow the `"geom"` column name in any outputs from any of the functions. As I understand, there is always just one geometry column for each `gpd.GeoDataFrame` that is outputted from the functions. I would therefore just use the standard `"geometry"` column name. If multiple geometry columns exist internally at some point, it is fine to differentiate between them, but the `"geom"` column should not be exposed outside the functions as it is confusing. I also encountered errors in the `cba_introduction.ipynb` where a `pd.DataFrame` had a `"geom"` column but was not a `GeoDataFrame`.

Reply:
I edited out the `"geom"` column. The only reference column for geometry is now `"geometry"` for all `GeoDataFrame`.